### ISSUE TITLE
[CH] Optimize get_json_object nested function calls

### ISSUE THREAD

**Issue by KevinyhZou** [apache/incubator-gluten#8304]
### Description

In our online sqls, have some nested `get_json_object` function calls, like `get_json_object(get_json_object(d, '$.a'), '$.b')`, and some of them can be optimized to `get_json_object(d, '$.a.b')`, which can improve the performance

![image](https://github.com/user-attachments/assets/bfdb5aa9-044b-48c8-a78c-b8d41bf13e5d)

![image](https://github.com/user-attachments/assets/dfc0e3f4-7dde-4feb-9447-8b45181a66bd)

**Comment by KevinyhZou** [comment]
some nested get_json_object call case can not optimized:  `select get_json_object(get_json_object('{"a":"{\\\"x\\\":5}"}', '$.a'), '$.x')` can not be converted to `select get_json_object('{"a":"{\\\"x\\\":5}"}', '$.a.x');`

**Comment by KevinyhZou** [comment]
Yes, it is a corner case. In most cases, it can be converted. @PHILO-HE

**Comment by PHILO-HE** [comment]
@KevinyhZou, I feel we can merge the patch for both CH backend and velox backend if it's in good shape. If you are afraid of any similar issue, it's ok to keep a config for users to control the use of this optimization. Thanks!
